### PR TITLE
Caching Doc: Create separate entry for opcache

### PR DESCRIPTION
**Pull request by danielzhang130**
The doc mixes APCu and Zend opcache in one entry. This is confusing.

This change clarifies that APCu is only a data cache and opcache is a opcode cache.

Note: Zend opcache is now part of PHP, so it's referred directly as opcache.

### CODE REVIEW

**Comment by mmattel** [comment]
Thanks for spotting this 👍 
I took the chance to do more updates of the document.

@EParzefall mind to do a in depth check 😄

File: modules/admin_manual/pages/configuration/server/caching_configuration.adoc [hunk 1]
```diff
@@ -7,83 +7,75 @@
...
+  Opcache is bundled with PHP from version 5.5.0 and later.
 * xref:apcu[APCu] +
-  APCu 4.0.6 and up is required. Alternatively you can use the Zend OPCache.
-  However, *it is not a data cache*, it is an opcode cache only.
+  This is a data cache only and does *not* cache any opcode.
+  APCu 4.0.6 and up is required.
 * xref:redis[Redis] +
-  This is a distributed cache for multi-server ownCloud installations and provides file locking.
-  Version 2.2.6 or higher of the PHP Redis extension is required.
+  This is an in-memory data structure store (cache) for single and multi-server ownCloud installations, provides file locking and can be setup in local or distributed environments. Consider Redis younger, more feature-rich and configurable than memchached. At minimum, avrsion 2.2.6 or higher of the PHP Redis extension is required.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/caching_configuration.adoc:36]:
```suggestion
  This is an in-memory data structure store (cache) for single and multi-server ownCloud installations, provides file locking and can be setup in local or distributed environments. Consider Redis younger, more feature-rich and configurable than memcached. At minimum, version 2.2.6 or higher of the PHP Redis extension is required.
```

File: modules/admin_manual/pages/configuration/server/caching_configuration.adoc [hunk 2]
```diff
@@ -7,83 +7,75 @@
...
 * xref:memcached[Memcached] +
-  This is a distributed cache for multi-server ownCloud installations.
+  This is a distributed cache for multi-server ownCloud installations and has *no* file locking capabilities.
+
+See following page to learn more about the {redis-memcached-url}[Redis vs. Memcached – 2021 Comparison].
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/caching_configuration.adoc:40]:
```suggestion
See the following page to learn more about the {redis-memcached-url}[Redis vs. Memcached – 2021 Comparison].
```

File: modules/admin_manual/pages/configuration/server/caching_configuration.adoc [hunk 3]
```diff
@@ -96,48 +88,37 @@ sudo apt install redis-server php-redis
...
 sudo redis-cli
 SELECT <dbIndex>
 FLUSHDB
 ----
 
-`<dbIndex>` is the number of Redis database where the cache is stored.
-It is zero by default at ownCloud. To check what yours is currently set
-to, check the `dbindex` value in `config/config.php`. Here’s an example
-of what to look for:
+`<dbIndex>` is the number of Redis database where the cache is stored. It is zero by default at ownCloud. To check what yours is currently set to, check the `dbindex` value in `config/config.php`. Here’s an example of what to look for:
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/caching_configuration.adoc:108]: Where is the example?

Comment (mmattel) [reply modules/admin_manual/pages/configuration/server/caching_configuration.adoc:108]: Thanks for spotting it 👍 
These things always happen on a bigger review of existing documents...

Comment (mmattel) [reply modules/admin_manual/pages/configuration/server/caching_configuration.adoc]: fixed

**Review by phil-davis — APPROVED** [review]

**Comment by mmattel** [comment]
Needs backporting to 10.7 and 10.8

File: modules/admin_manual/pages/configuration/server/caching_configuration.adoc [hunk 4]
```diff
@@ -168,9 +155,7 @@ After that, assuming that you don’t encounter any errors:
 
 ==== Clearing the Memcached Cache
 
-The Memcached cache can be flushed from the command-line using a range
-of common Linux/Unix tools, including `netcat` and `telnet`.
-The following example uses telnet to login, run the {flushall_url}[flush_all command], and logout:
+The Memcached cache can be flushed from the command-line using a range of common Linux/Unix tools, including `netcat` and `telnet`. The following example uses telnet to login, run the {flushall_url}[flush_all command], and logout:
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/server/caching_configuration.adoc:158]:
```suggestion
The Memcached cache can be flushed from the command line, using a range of common Linux/Unix tools including `netcat` and `telnet`. The following example uses telnet to log in, run the {flushall_url}[flush_all command], and log out:
```

File: modules/admin_manual/pages/configuration/server/caching_configuration.adoc [hunk 5]
```diff
@@ -236,8 +223,7 @@ The following example `config.php` configuration connects to a Redis cache via T
 
 ==== Redis Configuration Using Unix Sockets
 
-If Redis is running on the same server as ownCloud, it is recommended to configure it to use
-Unix sockets. Then, configure ownCloud to communicate with Redis, as in the following example. 
+If Redis is running on the same server as ownCloud, it is recommended to configure it to use Unix sockets. Then, configure ownCloud to communicate with Redis, as in the following example.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/server/caching_configuration.adoc:226]:
```suggestion
If Redis is running on the same server as ownCloud, it is recommended to configure it to use Unix sockets. Then, configure ownCloud to communicate with Redis as in the following example. 
```

File: modules/admin_manual/pages/configuration/server/caching_configuration.adoc [hunk 6]
```diff
@@ -447,18 +424,13 @@ which uses Redis as the cache backend:
...
-See {redis_security_url} for more information.
+CAUTION: For enhanced security, it is recommended to configure Redis to require a password. See {redis_security_url} for more information.
 
 == Caching Exceptions
 
-If ownCloud is configured to use either Memcached or Redis as a memory
-cache, please be aware that you may encounter issues with functionality.
-When these occur, it is usually a result of PHP being incorrectly
-configured, or the relevant PHP extension not being available.
+If ownCloud is configured to use either Memcached or Redis as a memory cache, please be aware that you may encounter issues with functionality. When these occur, it is usually a result of PHP being incorrectly configured, or the relevant PHP extension not being available.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/server/caching_configuration.adoc:431]:
```suggestion
If ownCloud is configured to use either Memcached or Redis as a memory cache, you may encounter issues with functionality. When these occur, it is usually a result of PHP being incorrectly configured or the relevant PHP extension not being available.
```

**Review by EParzefall — APPROVED** [review]
Minor language/clarity issues.

**Comment by mmattel** [comment]
squashed and rebased

File: modules/admin_manual/pages/configuration/server/caching_configuration.adoc [hunk 7]
```diff
@@ -7,83 +7,80 @@
...
 
 == Cache Types
 
+=== Opcache
+
+Opcache should be enabled by default in your php installation. To check it, run the following command:
+
+[source,php]
+----
+php -r 'phpinfo();' | grep opcache.enable
```
Comment (danielzhang130) [inline modules/admin_manual/pages/configuration/server/caching_configuration.adoc:59]: Slight caveat that opcache may be enabled on webserver and not on cli

**Pull request merged**